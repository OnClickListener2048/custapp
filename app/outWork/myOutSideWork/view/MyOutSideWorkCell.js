/**
 * 我的外勤item布局
 * Created by jiaxueting on 2017/6/14.
 */

import React,{Component,PropTypes}from 'react';
import styles from "../css/MyOutSideWorkStyle";
import {Image, Text, View} from "react-native";


export default class MessageCenterCell extends Component {
    constructor(props) {
        super(props)
        this.state = {
            statusIcon: this.props.statusIcon,
            statusName: this.props.statusName,
            companyName: this.props.companyName,
            statusContent: this.props.statusContent,
            statusCourse: this.props.statusCourse,

        }
    }

    static propTypes = {
        statusIcon: PropTypes.string,
        statusName: PropTypes.string,
        companyName: PropTypes.string,
        statusContent: PropTypes.string,
        statusCourse: PropTypes.string,
    };

    // _statusCourse(){
    //     if(this.state.statusCourse==0){
    //         return '待处理'
    //     }else if(this.state.statusCourse==1){
    //         return '进行中'
    //     }else if(this.state.statusCourse==2){
    //         return '已完成'
    //     }else if(this.state.statusCourse==3){
    //         return '已取消'
    //     }
    // }

    _statusCourseColor(){
        if(this.props.statusCourse=='待处理'||this.props.statusCourse=='待分配'){
            return 'orange'
        }else if(this.props.statusCourse=='进行中'){
            return 'green'
        }else if(this.props.statusCourse=='已完成'){
            return 'red'
        }else if(this.props.statusCourse=='已取消'){
            return '#e6e6e6'
        }
    }

    render() {
        const {statusIcon,statusName,companyName,statusContent,statusCourse} = this.props
        return (
            <View
                style={styles.rowStyle}>
                <View
                    style={styles.realRowStyle}>
                    <Image
                        source={require('../../../img/bigk.png')}
                        style={[styles.messageBottomStyle]}>
                    <View
                        style={styles.titleViewStyle}>
                        <Text
                            textAlign='left'
                            numberOfLines={1}
                            style={[{fontSize: 15,marginTop: 15, marginLeft : 20 , color : this.props.statusCourse=='已取消'?'#e6e6e6':'#323232'}] }>{statusName}</Text>
                        <Text
                            textAlign='left'
                            numberOfLines={1}
                            style={[{fontSize: 14,marginTop: 10, marginLeft :20 , color : this.props.statusCourse=='已取消'?'#e6e6e6':'#969696'}] }>{companyName}</Text>
                    </View>
                    <Text
                        textAlign='center'
                        style={[styles.timeTitleStyle,{marginRight:20,color:this.props.statusCourse=='已取消'?'#e6e6e6':'#323232'}]}>{statusContent}</Text>

                    <Text
                        textAlign='right'
                        style={[styles.timeTitleStyle,{color:this.props.statusCourse=='已取消'?'#e6e6e6':'#a2a1a6'}]}>{statusCourse}</Text>
                    </Image>
                </View>
                <View style={[styles.badgeBubble,
                    {backgroundColor: this._statusCourseColor()}]}>
                    <Text style={styles.badgeText}>{statusIcon}</Text>
                </View>
            </View>
        )
    }
}