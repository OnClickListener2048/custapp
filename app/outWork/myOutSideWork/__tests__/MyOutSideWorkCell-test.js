/**
 * Created by liufei on 2017/8/23.
 */

import 'react-native';
import React from 'react';
import MyOutSideWorkCell from '../view/MyOutSideWorkCell'
import renderer from 'react-test-renderer';

test('快照正确渲染', () => {
    const tree = renderer.create(
        <MyOutSideWorkCell  navigator={navigator}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    // 只能操作属性
    expect(tree).toMatchSnapshot();
});

