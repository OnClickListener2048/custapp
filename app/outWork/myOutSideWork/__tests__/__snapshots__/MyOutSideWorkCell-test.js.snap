// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照正确渲染 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flexDirection": "column",
      "height": 75,
      "marginTop": 18,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#FAFAFA",
        "borderRadius": 6,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 75,
        "marginLeft": 23.5,
        "marginRight": 15,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 75,
            "resizeMode": "stretch",
            "width": 707.5,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "height": 65,
            "marginLeft": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#323232",
                "fontSize": 15,
                "marginLeft": 20,
                "marginTop": 15,
              },
            ]
          }
          textAlign="left"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#969696",
                "fontSize": 14,
                "marginLeft": 20,
                "marginTop": 10,
              },
            ]
          }
          textAlign="left"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "color": "#a2a1a6",
              "fontSize": 14,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 15,
              "marginTop": 30,
            },
            Object {
              "color": "#323232",
              "marginRight": 20,
            },
          ]
        }
        textAlign="center"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "color": "#a2a1a6",
              "fontSize": 14,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 15,
              "marginTop": 30,
            },
            Object {
              "color": "#a2a1a6",
            },
          ]
        }
        textAlign="right"
      />
    </Image>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 37,
          "justifyContent": "center",
          "marginLeft": 7,
          "marginTop": 18,
          "position": "absolute",
          "width": 37,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "top": -0.5,
        }
      }
    />
  </View>
</View>
`;

exports[`快照正确渲染 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flexDirection": "column",
      "height": 75,
      "marginTop": 18,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#FAFAFA",
        "borderRadius": 6,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 75,
        "marginLeft": 23.5,
        "marginRight": 15,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 75,
            "resizeMode": "stretch",
            "width": 707.5,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "height": 65,
            "marginLeft": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#323232",
                "fontSize": 15,
                "marginLeft": 20,
                "marginTop": 15,
              },
            ]
          }
          textAlign="left"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#969696",
                "fontSize": 14,
                "marginLeft": 20,
                "marginTop": 10,
              },
            ]
          }
          textAlign="left"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "color": "#a2a1a6",
              "fontSize": 14,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 15,
              "marginTop": 30,
            },
            Object {
              "color": "#323232",
              "marginRight": 20,
            },
          ]
        }
        textAlign="center"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "color": "#a2a1a6",
              "fontSize": 14,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 15,
              "marginTop": 30,
            },
            Object {
              "color": "#a2a1a6",
            },
          ]
        }
        textAlign="right"
      />
    </Image>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 37,
          "justifyContent": "center",
          "marginLeft": 7,
          "marginTop": 18,
          "position": "absolute",
          "width": 37,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "top": -0.5,
        }
      }
    />
  </View>
</View>
`;
